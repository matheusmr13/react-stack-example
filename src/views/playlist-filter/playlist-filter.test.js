import React from 'react';
import { shallow } from 'enzyme';
import EntityFilter from 'components/entity-filter';
import PlaylistFilter from './index';

const filters = [{
	id: 'locale',
	name: 'Locale',
	values: [{
		value: 'en_AU',
		name: 'en_AU'
	}, {
		value: 'de_DE',
		name: 'de_DE '
	}]
}, {
	id: 'timestamp',
	name: 'Data e Horário',
	validation: {
		primitiveType: 'STRING',
		entityType: 'DATE_TIME',
		pattern: 'yyyy-MM-ddTHH:mm:ss'
	}
}, {
	id: 'limit',
	name: 'Quantidade',
	validation: {
		primitiveType: 'INTEGER',
		min: 1,
		max: 50
	}
}, {
	id: 'offset',
	name: 'Página',
	validation: {
		primitiveType: 'INTEGER'
	}
}];

describe('PlaylistFilter component', () => {
	const fetchFilters = jest.fn();
	const onFilterChange = jest.fn();
	const playlistList = shallow(<PlaylistFilter
		fetchFilters={fetchFilters}
		onFilterChange={onFilterChange}
		loadingFilters
		selectedFilters={{}}
		possibleFilters={[]}
	/>);
	it('should render loading properly', () => {
		expect(playlistList).toMatchSnapshot();
	});
	it('should render filters properly', () => {
		playlistList.setProps({
			fetchFilters,
			loadingFilters: false,
			selectedFilters: {},
			possibleFilters: filters
		});
		expect((playlistList).find(EntityFilter)).toHaveLength(1);
		expect(fetchFilters).toHaveBeenCalled();
		expect(playlistList).toMatchSnapshot();
	});
});
