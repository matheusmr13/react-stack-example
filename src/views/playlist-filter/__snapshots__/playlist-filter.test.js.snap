// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistFilter component should render filters properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    fetchFilters={[Function]}
    loadingFilters={false}
    onFilterChange={[Function]}
    possibleFilters={
        Array [
            Object {
              "id": "locale",
              "name": "Locale",
              "values": Array [
                Object {
                  "name": "en_AU",
                  "value": "en_AU",
                },
                Object {
                  "name": "de_DE ",
                  "value": "de_DE",
                },
              ],
            },
            Object {
              "id": "timestamp",
              "name": "Data e Horário",
              "validation": Object {
                "entityType": "DATE_TIME",
                "pattern": "yyyy-MM-ddTHH:mm:ss",
                "primitiveType": "STRING",
              },
            },
            Object {
              "id": "limit",
              "name": "Quantidade",
              "validation": Object {
                "max": 50,
                "min": 1,
                "primitiveType": "INTEGER",
              },
            },
            Object {
              "id": "offset",
              "name": "Página",
              "validation": Object {
                "primitiveType": "INTEGER",
              },
            },
          ]
    }
    selectedFilters={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <EntityFilter
        filters={
                Array [
                        Object {
                          "id": "locale",
                          "name": "Locale",
                          "values": Array [
                            Object {
                              "name": "en_AU",
                              "value": "en_AU",
                            },
                            Object {
                              "name": "de_DE ",
                              "value": "de_DE",
                            },
                          ],
                        },
                        Object {
                          "id": "timestamp",
                          "name": "Data e Horário",
                          "validation": Object {
                            "entityType": "DATE_TIME",
                            "pattern": "yyyy-MM-ddTHH:mm:ss",
                            "primitiveType": "STRING",
                          },
                        },
                        Object {
                          "id": "limit",
                          "name": "Quantidade",
                          "validation": Object {
                            "max": 50,
                            "min": 1,
                            "primitiveType": "INTEGER",
                          },
                        },
                        Object {
                          "id": "offset",
                          "name": "Página",
                          "validation": Object {
                            "primitiveType": "INTEGER",
                          },
                        },
                      ]
        }
        onChange={[Function]}
        values={Object {}}
/>,
      "style": Object {
        "padding": "16px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "filters": Array [
          Object {
            "id": "locale",
            "name": "Locale",
            "values": Array [
              Object {
                "name": "en_AU",
                "value": "en_AU",
              },
              Object {
                "name": "de_DE ",
                "value": "de_DE",
              },
            ],
          },
          Object {
            "id": "timestamp",
            "name": "Data e Horário",
            "validation": Object {
              "entityType": "DATE_TIME",
              "pattern": "yyyy-MM-ddTHH:mm:ss",
              "primitiveType": "STRING",
            },
          },
          Object {
            "id": "limit",
            "name": "Quantidade",
            "validation": Object {
              "max": 50,
              "min": 1,
              "primitiveType": "INTEGER",
            },
          },
          Object {
            "id": "offset",
            "name": "Página",
            "validation": Object {
              "primitiveType": "INTEGER",
            },
          },
        ],
        "onChange": [Function],
        "values": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <EntityFilter
          filters={
                    Array [
                              Object {
                                "id": "locale",
                                "name": "Locale",
                                "values": Array [
                                  Object {
                                    "name": "en_AU",
                                    "value": "en_AU",
                                  },
                                  Object {
                                    "name": "de_DE ",
                                    "value": "de_DE",
                                  },
                                ],
                              },
                              Object {
                                "id": "timestamp",
                                "name": "Data e Horário",
                                "validation": Object {
                                  "entityType": "DATE_TIME",
                                  "pattern": "yyyy-MM-ddTHH:mm:ss",
                                  "primitiveType": "STRING",
                                },
                              },
                              Object {
                                "id": "limit",
                                "name": "Quantidade",
                                "validation": Object {
                                  "max": 50,
                                  "min": 1,
                                  "primitiveType": "INTEGER",
                                },
                              },
                              Object {
                                "id": "offset",
                                "name": "Página",
                                "validation": Object {
                                  "primitiveType": "INTEGER",
                                },
                              },
                            ]
          }
          onChange={[Function]}
          values={Object {}}
/>,
        "style": Object {
          "padding": "16px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filters": Array [
            Object {
              "id": "locale",
              "name": "Locale",
              "values": Array [
                Object {
                  "name": "en_AU",
                  "value": "en_AU",
                },
                Object {
                  "name": "de_DE ",
                  "value": "de_DE",
                },
              ],
            },
            Object {
              "id": "timestamp",
              "name": "Data e Horário",
              "validation": Object {
                "entityType": "DATE_TIME",
                "pattern": "yyyy-MM-ddTHH:mm:ss",
                "primitiveType": "STRING",
              },
            },
            Object {
              "id": "limit",
              "name": "Quantidade",
              "validation": Object {
                "max": 50,
                "min": 1,
                "primitiveType": "INTEGER",
              },
            },
            Object {
              "id": "offset",
              "name": "Página",
              "validation": Object {
                "primitiveType": "INTEGER",
              },
            },
          ],
          "onChange": [Function],
          "values": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PlaylistFilter component should render loading properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    fetchFilters={[Function]}
    loadingFilters={true}
    onFilterChange={[Function]}
    possibleFilters={Array []}
    selectedFilters={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Loading",
  Symbol(enzyme.__nodes__): Array [
    "Loading",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
